src/js/plugins/search/view.js

```diff
@@ -122,6 +126,102 @@ class SearchView {
+          Results
+        </h4>
+        <div class="pagination">
+          <a href="#" class="prev" title="Previous result"><span class="icon icon-chevron hflip"></span></a>
+          <span data-id="resultsCount">${this.resultsPosition()}</span>
+          <a href="#" class="next" title="Next result"><span class="icon icon-chevron"></a>
+        </div>
+        <a href="#" class="clear" title="Clear search results">
+          <span class="icon icon-close"></span>
+        </a>
+      </div>
+    `);
+    this.dom.searchNavigation = $(`.${selector}`);
+  }
+
+  resultsPosition() {
+    let positionMessage = `${this.matches.length} result${this.matches.length === 1 ? '' : 's'}`;
+    if (~this.currentMatchIndex) {
+      positionMessage = `${this.currentMatchIndex + 1} / ${this.matches.length}`;
+    }
+    return positionMessage;
+  }
+
+  bindSearchNavigationEvents() {
+    if (!this.dom.searchNavigation) { return; }
+    const namespace = 'searchNavigation';
+
+    this.dom.searchNavigation
+      .on(`click.${namespace}`, '.clear', this.clearSearchFieldAndResults.bind(this))
+      .on(`click.${namespace}`, '.prev', this.showPrevResult.bind(this))
+      .on(`click.${namespace}`, '.next', this.showNextResult.bind(this))
+      .on(`click.${namespace}`, false);
+  }
+
+  showPrevResult() {
+    if (this.currentMatchIndex === 0) { return; }
+    if (this.br.mode === this.br.constModeThumb) { this.br.switchMode(this.br.constMode1up); }
+    if (!~this.currentMatchIndex) { this.currentMatchIndex = 1; }
+    this.br.$('.BRnavline .BRsearch').eq(--this.currentMatchIndex).click();
+    this.updateResultsPosition();
+  }
+
+  showNextResult() {
+    if (this.currentMatchIndex + 1 === this.matches.length) { return; }
+    if (this.br.mode === this.br.constModeThumb) { this.br.switchMode(this.br.constMode1up); }
+    this.br.$('.BRnavline .BRsearch').eq(++this.currentMatchIndex).click();
+    this.updateResultsPosition();
+  }
+
+  updateResultsPosition() {
+    this.dom.searchNavigation.find('[data-id=resultsCount]').text(this.resultsPosition());
+  }
+
+  teardownSearchNavigation() {
+    if (!this.dom.searchNavigation) { return; }
+
+    this.dom.searchNavigation.off('.searchNavigation').remove();
+    this.dom.searchNavigation = null;
+  }
+
+  setCurrentMatchIndex() {
+    let matchingSearchResult;
+    if (this.br.mode === this.br.constModeThumb) {
+      this.currentMatchIndex = -1;
+      return;
+    }
+    if (this.br.mode === this.br.constMode2up) {
+      matchingSearchResult = this.find2upMatchingSearchResult();
+    }
+    else {
+      matchingSearchResult = this.find1upMatchingSearchResult();
+    }
+    this.currentMatchIndex = this.matches.indexOf(matchingSearchResult);
+  }
+
+  find1upMatchingSearchResult() {
+    return this.matches.find((m) => this.br.currentIndex() === m.par[0].page - 1);
+  }
+
+  find2upMatchingSearchResult() {
+    return this.matches.find((m) => this.br._isIndexDisplayed(m.par[0].page - 1));
+  }
```

Comment:
QUESTION: For either of these functions, will `m.par[0].page` ever be 0? Is that okay given we're subtracting 1 from the value?

Reply:
The page number is the human-readable number, not the index. Since internally BookReader is only concerned with the index of the page and its data, I need to subtract one from the value.